platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/problems/ConfigurationCacheProblemsSummary.kt

```diff
@@ -120,16 +120,16 @@ class ConfigurationCacheProblemsSummary(
             if (severity != ProblemSeverity.SuppressedSilently) {
                 val isNewCause = recordProblemCause(problem, severity)
                 if (isNewCause) {
-                    collectOriginalException(problem)
+                    collectOriginalException(problem, severity)
                 }
             }
             return true
         }
     }
 
     private
-    fun collectOriginalException(problem: PropertyProblem) {
-        if (originalProblemExceptions.size < MAX_PROBLEM_EXCEPTIONS) {
+    fun collectOriginalException(problem: PropertyProblem, severity: ProblemSeverity) {
+        if (originalProblemExceptions.size < MAX_PROBLEM_EXCEPTIONS && severity != ProblemSeverity.Interrupting) {
```

Comment:
💭  I purposefully placed this check into the method instead of putting it at the call-site, since some condition checking has already been happening here, and it's important that we still collect causes of the interrupting problems (aka calling `recordProblemCause`). I wanted to avoid questions or confusion as to why it's not merged with the check for the `SuppressedSilently` severity.